sled-agent/src/bin/sled-agent-sim.rs

```diff
@@ -76,6 +77,7 @@ async fn do_run() -> Result<(), CmdError> {
             ip: (*args.sled_agent_addr.ip()).into(),
         },
         updates: ConfigUpdates { zone_artifact_path: tmp.path().to_path_buf() },
+        hardware: ConfigHardware { hardware_threads: 4, physical_ram: 1 << 30 },
```

Comment:
I think these should be more generous (e.g. matching what's in test-utils). With this config, a sled-agent-sim sled can have exactly one instance with 1 GiB of RAM, since we require guest memory to be specified in 1 GiB increments.

Reply:
Done in ec2d3f82bd1bbd9999e2512140d68e9036ba775f